### ISSUE TITLE
Azure Batch: DeleteJob(async) fails

### ISSUE THREAD

**Issue by ChristianSauer** [Azure/azure-sdk-for-net#4602]
Hi,
I am currently using Docker Linux Containers ( microsoft/dotnet:2.1.1-aspnetcore-runtime) aund Azure Batch 8.1.2 and I am trying to delete an job like this:

```
  using (var bc = await BatchClient.OpenAsync(sharedKeyCredentials))
                {
                                     bc.JobOperations.DeleteJob(finishedJob);
                        }
```

The job exists, but I just get an BatchException which contains this message:
"Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nRequestId:fc2efe54-9b9e-441d-a9bf-3377d55fda88\nTime:2018-07-30T09:36:45.2244840Z"

When I delete this job through BatchLabs, there is no issue.

**Comment by matthchr** [comment]
Hi @ChristianSauer - we're aware of this issue. It's due to how we implemented shared key auth and [this](https://github.com/dotnet/corefx/issues/31172) issue in netcoreapp2.1

For now we're recommending folks target netcoreapp2.0. We will release a fix to this in the upcoming 9.0 release in the next few weeks.

**Comment by takekazuomi** [comment]
Workaround. It's works for me.
> You can also use an environment variable to opt out of using sockets implementations based on SocketsHttpHandler. To do this, set the DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER to either false or 0.
https://docs.microsoft.com/en-us/dotnet/core/whats-new/dotnet-core-2-1#sockets-improvements
